### PR TITLE
Update go-oidc to v3

### PR DESCRIPTION
**Pull request by hcsu**
Update go-oidc to v3

Ref: https://github.com/coreos/go-oidc#updates-from-v2-to-v3

Seems old package already not working, do we need update oidc to v3 here?

### CODE REVIEW

**Comment by geckofu** [comment]
@hcsu Thanks for submitting this patch. 
Please also update the import path in `main.go`, and you can test by building the binary locally to see if there are any more errors.

File: main.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ import (
 	awscreds "github.com/aws/aws-sdk-go/aws/credentials"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/sts"
-	"github.com/coreos/go-oidc"
+	"github.com/coreos/go-oidc/v3/oidc"
```
Comment (hcsu) [inline main.go:26]: Hi @geckofu , thanks for review.
> Please also update the import path in main.go

means update here?

I tried build updated version, and I can login Console and get AWS STS token successfully.

```
oidc2aws git:update-go-oidc-v3✗ ➜ ./main -login -alias mcd-admin
oidc2aws git:update-go-oidc-v3✗ ➜ ./main -env -alias mcd-admin
export AWS_ACCESS_KEY_ID=ASIA5UHZSFX4BQM...
export AWS_SECRET_ACCESS_KEY=GnytU+xIN8IBW+sewxbc...
export AWS_SESSION_TOKEN=FwoGZXIvYXdzEF8aD...
oidc2aws git:update-go-oidc-v3✗ ➜
```
Btw seems it's already not working if we don't update this breaking change.

Before update:
```
theplant ➜ go get -u -v github.com/theplant/oidc2aws
github.com/theplant/oidc2aws (download)
github.com/BurntSushi/toml (download)
github.com/aws/aws-sdk-go (download)
github.com/jmespath/go-jmespath (download)
github.com/coreos/go-oidc (download)
package github.com/coreos/go-oidc: no Go files in /Users/sunhaochuan/go/src/github.com/coreos/go-oidc
github.com/pkg/errors (download)
get "golang.org/x/oauth2": found meta tag get.metaImport{Prefix:"golang.org/x/oauth2", VCS:"git", RepoRoot:"https://go.googlesource.com/oauth2"} at //golang.org/x/oauth2?go-get=1
golang.org/x/oauth2 (download)
get "golang.org/x/oauth2/internal": found meta tag get.metaImport{Prefix:"golang.org/x/oauth2", VCS:"git", RepoRoot:"https://go.googlesource.com/oauth2"} at //golang.org/x/oauth2/internal?go-get=1
get "golang.org/x/oauth2/internal": verifying non-authoritative meta tag
get "golang.org/x/net/context/ctxhttp": found meta tag get.metaImport{Prefix:"golang.org/x/net", VCS:"git", RepoRoot:"https://go.googlesource.com/net"} at //golang.org/x/net/context/ctxhttp?go-get=1
get "golang.org/x/net/context/ctxhttp": verifying non-authoritative meta tag
golang.org/x/net (download)
theplant ➜
```

**Review by hcsu — COMMENTED** [review]
> Please also update the import path in main.go, and you can test by building the binary locally to see if there are any more errors.

Thanks, please refer to below discussion.

**Comment by bodhi** [comment]
@hcsu what's happening with this PR?

**Comment by hcsu** [comment]
> @hcsu what's happening with this PR?

Several month ago, I re-installed my Mac so I need re-install oidc2aws as well, then I noticed dependency `github.com/coreos/go-oidc` seems out of date (ref: https://github.com/coreos/go-oidc/tree/v3#updates-from-v2-to-v3), so I raised this PR for upgrade `go-oidc` v3 (`v3` is the default branch of `github.com/coreos/go-oidc`). 

Gecko will help to review when he when free time.

**Comment by hcsu** [comment]
Since this PR is not up to date, so I'm going to close it, will create a new PR if needed.

**Pull request closed without merging**